util/ip.go

```diff
@@ -0,0 +1,26 @@
+package util
+
+import (
+  "fmt"
+  "io/ioutil"
+  "net/http"
+)
+
+func GetIPAdress() (string, error) { // TODO wonering if there is a cleaner way to get public ip.
+  url := "https://api.ipify.org?format=text"	// we are using a pulib IP API, we're using ipify here, below are some others
```

Comment:
Maybe the provider for public address should be optional? Maybe not now, but in the future - have the public IP provider come from a config file, maybe use this as the default or backup/fallback option.

Reply:
Interesting, thought. I think that overall, the config file would be more of a headache for the user. However, giving the option may allow for more flexibility, a good future addition!